### ISSUE TITLE
NETCUP: UTF-8 (non-punycode) not supported?

### ISSUE THREAD

**Issue by killerbees19** [StackExchange/dnscontrol#2451]
It seems IDN domains are only supported **with** punycode format at v4.1.1.

* `D("exämple.at", …` tries to (re-) create all records and silently fails. (They are not visible at the netcup CCP.)
* `D("xn--exmple-cua.at", …` works as expected.

/cc @kordianbruck

**Comment by patschi** [comment]
Just for some context: Same issue has been raised for provider INWX too, with issue #2450

**Comment by kordianbruck** [comment]
Yeah, seemingly there needs to be a decision what the input format for dnscontrol should be. I think Tom's suggestion there is reasonable and would address this issue.

**Comment by tlimoncelli** [comment]
(I just had this thought... it may be totally wrong but I'll put it out there)

Proposal:

Domains: The goal should be that the string a user enters in dnsconfig.js should match the string they see in the provider's web portal.  That would simplify things for the user, and give us an obvious goal for how the code is supposed to work.

Records: The individual records should permit punycode and UTF-8, and be converted to the format that the provider prefers.  This could be all UTF-8, all punycode, or a mix (use whatever string is in dnsconfig.js).

Would that work?

**Comment by tlimoncelli** [comment]
Related: https://github.com/StackExchange/dnscontrol/issues/2097

**Comment by kordianbruck** [comment]
I think for now I'll add a check to at least alert the user to undefined behavior and possibly fail the actuation.
